Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp

```diff
@@ -0,0 +1,1185 @@
+            else {
+                bool validSize = WTF::switchOn(WTFMove(*size),
+                    [&](CSSRadialGradientValue::ExtentKeyword&& extent) -> bool {
+                        data.gradientBox = CSSRadialGradientValue::EllipseOfExtent { WTFMove(extent), WTFMove(position) };
+                        return true;
+                    },
+                    [&](Ref<CSSPrimitiveValue>&&) -> bool {
+                        // Ellipses must have two length-percentages specified.
+                        return false;
+                    },
+                    [&](std::pair<Ref<CSSPrimitiveValue>, Ref<CSSPrimitiveValue>>&& size) -> bool {
+                        data.gradientBox = CSSRadialGradientValue::EllipseOfSize { WTFMove(size), WTFMove(position) };
+                        return true;
+                    }
+                );
+                if (!validSize)
+                    return nullptr;
+            }
+            break;
+        }
+    } else {
+        if (!size) {
+            if (position)
+                data.gradientBox = CSSRadialGradientValue::GradientBox { std::in_place_type<CSSGradientPosition>, WTFMove(*position) };
+            else
+                data.gradientBox = std::monostate { };
+        } else {
+            WTF::switchOn(WTFMove(*size),
+                [&](CSSRadialGradientValue::ExtentKeyword&& extent) {
+                    data.gradientBox = CSSRadialGradientValue::Extent { WTFMove(extent), WTFMove(position) };
+                },
+                [&](Ref<CSSPrimitiveValue>&& length) {
+                    data.gradientBox = CSSRadialGradientValue::Length { WTFMove(length), WTFMove(position) };
+                },
+                [&](std::pair<Ref<CSSPrimitiveValue>, Ref<CSSPrimitiveValue>>&& size) {
+                    data.gradientBox = CSSRadialGradientValue::Size { WTFMove(size), WTFMove(position) };
+                }
+            );
+        }
+    }
+
+    return CSSRadialGradientValue::create(
+        WTFMove(data),
+        repeating,
+        computedColorInterpolationMethod,
+        WTFMove(*stops)
+    );
+}
+
+// MARK: <conic-gradient()> | <repeating-conic-gradient()>
+// https://drafts.csswg.org/css-images-4/#conic-gradient-syntax
+
+static RefPtr<CSSValue> consumeConicGradient(CSSParserTokenRange& range, const CSSParserContext& context, CSSGradientRepeat repeating)
+{
+    // conic-gradient() = conic-gradient(
+    //   [ [ from <angle> ]? [ at <position> ]? ] || <color-interpolation-method>,
+    //   <angular-color-stop-list>
+    // )
+
+    std::optional<ColorInterpolationMethod> colorInterpolationMethod;
+
+    if (range.peek().id() == CSSValueIn) {
+        colorInterpolationMethod = consumeColorInterpolationMethod(range);
+        if (!colorInterpolationMethod)
+            return nullptr;
+    }
+
+    CSSConicGradientValue::Angle angle = std::monostate { };
```

Comment:
Is `= std::monostate { }` needed? Why isn’t that what the default constructor makes?